spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -16,164 +16,49 @@
-	 * Kubernetes API server.
-	 */
-	private String filter;
-
-	/** Set the port numbers that are considered secure and use HTTPS. */
-	private Set<Integer> knownSecurePorts = Stream.of(443, 8443).collect(Collectors.toCollection(HashSet::new));
-
-	/**
-	 * If set, then only the services matching these labels will be fetched from the
-	 * Kubernetes API server.
-	 */
-	private Map<String, String> serviceLabels = new HashMap<>();
-
-	/**
-	 * If set then the port with a given name is used as primary when multiple ports are
-	 * defined for a service.
-	 */
-	private String primaryPortName;
-
-	private Metadata metadata = new Metadata();
-
-	private int order = DEFAULT_ORDER;
-
-	public boolean isEnabled() {
-		return this.enabled;
-	}
-
-	public void setEnabled(boolean enabled) {
-		this.enabled = enabled;
-	}
-
-	public String getFilter() {
-		return this.filter;
-	}
-
-	public void setFilter(String filter) {
-		this.filter = filter;
-	}
-
-	public Set<Integer> getKnownSecurePorts() {
-		return this.knownSecurePorts;
-	}
-
-	public void setKnownSecurePorts(Set<Integer> knownSecurePorts) {
-		this.knownSecurePorts = knownSecurePorts;
-	}
-
-	public Map<String, String> getServiceLabels() {
-		return this.serviceLabels;
-	}
-
-	public void setServiceLabels(Map<String, String> serviceLabels) {
-		this.serviceLabels = serviceLabels;
-	}
-
-	public String getPrimaryPortName() {
-		return primaryPortName;
-	}
-
-	public void setPrimaryPortName(String primaryPortName) {
-		this.primaryPortName = primaryPortName;
-	}
-
-	public Metadata getMetadata() {
-		return this.metadata;
-	}
-
-	public void setMetadata(Metadata metadata) {
-		this.metadata = metadata;
-	}
-
-	public boolean isAllNamespaces() {
-		return allNamespaces;
-	}
-
-	public void setAllNamespaces(boolean allNamespaces) {
-		this.allNamespaces = allNamespaces;
-	}
-
-	public boolean isIncludeNotReadyAddresses() {
-		return includeNotReadyAddresses;
-	}
-
-	public void setIncludeNotReadyAddresses(boolean includeNotReadyAddresses) {
-		this.includeNotReadyAddresses = includeNotReadyAddresses;
-	}
-
-	public int getOrder() {
-		return this.order;
-	}
-
-	public void setOrder(int order) {
-		this.order = order;
-	}
-
-	public boolean isWaitCacheReady() {
-		return waitCacheReady;
-	}
-
-	public void setWaitCacheReady(boolean waitCacheReady) {
-		this.waitCacheReady = waitCacheReady;
-	}
-
-	public long getCacheLoadingTimeoutSeconds() {
-		return cacheLoadingTimeoutSeconds;
-	}
-
-	public void setCacheLoadingTimeoutSeconds(long cacheLoadingTimeoutSeconds) {
-		this.cacheLoadingTimeoutSeconds = cacheLoadingTimeoutSeconds;
-	}
-
-	@Override
-	public String toString() {
-		return new ToStringCreator(this).append("enabled", this.enabled).append("filter", this.filter)
-				.append("knownSecurePorts", this.knownSecurePorts).append("serviceLabels", this.serviceLabels)
-				.append("metadata", this.metadata).toString();
+public record KubernetesDiscoveryProperties(
```

Comment:
this is the idea of the PR, simplify this to a record. everything else are just tests.